app/src/main/java/com/example/criminalintentapp/presentation/authentication/FirebaseAuthServiceImpl.kt

```diff
@@ -0,0 +1,48 @@
+package com.example.criminalintentapp.presentation.authentication
+
+import android.util.Log
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import kotlinx.coroutines.tasks.await
+
+class FirebaseAuthServiceImpl : FirebaseAuthService {
+
+    override suspend fun login(email: String, password: String): Result<Exception, FirebaseUser> {
+        Log.d("Service", "Login success")
```

Comment:
Since this tag is used 2 times, I suggest defining it as a `private const val TAG = "FirebaseAuthService" and then use it when logging events. Also, the message "login success" is shown before actually attempting to login, thus lying about the true state of the authentication. Please consider moving it somewhere else, or removing it all together. Either way is fine